symphony-bdk-legacy/sms-sdk-renderer-java/src/test/java/it/services/SmsRendererTest.java

```diff
@@ -0,0 +1,152 @@
+    this.testJsonArray(SmsRenderer.SmsTypes.TABLE);
+    this.testJsonArray(SmsRenderer.SmsTypes.INFORMATION);
+    this.testJsonArray(SmsRenderer.SmsTypes.NOTIFICATION);
+    this.testJsonArray(SmsRenderer.SmsTypes.ALERT);
+    this.testJsonArray(SmsRenderer.SmsTypes.SIMPLE);
+  }
+
+  // Private methods
+  private void testJsonArray(final SmsRenderer.SmsTypes smsTypes) throws IOException, ParseException {
+    assertNotNull(smsTypes);
+
+    final String typeName = smsTypes.getName().toLowerCase();
+
+    String jsonMessageContext = null;
+    JSONArray jsonArray = null;
+
+    if (smsTypes != SmsRenderer.SmsTypes.TABLE) {
+      final String fileName = typeName + "JsonArray.json";
+      jsonMessageContext = this.getStringFromFile("/SmsRenderer/JSONData/" + fileName);
+      jsonArray = new JSONArray();
+      jsonArray.add(jsonMessageContext);
+    } else {
+      final JSONParser parser = new JSONParser();
+      final String sourceFile = System.getProperty("user.dir") + "/src/test/resources/SmsRenderer/JSONData/tableJsonArray.json";
+      final Object obj = parser.parse(new FileReader(sourceFile));
+      final JSONObject jsonObject = (JSONObject) obj;
+      jsonArray = (JSONArray) jsonObject.get("phones");
+    }
+
+    final String compiledTemplate = this.minifyHTML(SmsRenderer.renderInBot(jsonArray, smsTypes));
+    assertNotNull(compiledTemplate);
+
+    this.verifyResultJSON(typeName, compiledTemplate, "JsonArrayHtml.html");
+  }
+
+  private void verifyResultJSON(final String typeName, final String compiledTemplate, final String fileName) throws IOException {
+    final String expectedResult = this.minifyHTML(this.getStringFromFile("/SmsRenderer/HTMLResult/" + typeName + fileName));
+    assertNotNull(expectedResult);
+    assertEquals(expectedResult, compiledTemplate);
+  }
+
+  private void testString(final SmsRenderer.SmsTypes smsTypes) throws IOException {
+    assertNotNull(smsTypes);
+
+    final String typeName = smsTypes.getName().toLowerCase();
+    final String jsonFileName = typeName + "Json.json";
+    final String jsonMessageContext = this.getStringFromFile("/SmsRenderer/JSONData/" + jsonFileName);
+
+    final String compiledTemplate = this.minifyHTML(SmsRenderer.renderInBot(jsonMessageContext, smsTypes));
+    assertNotNull(compiledTemplate);
+
+    this.verifyResultJSON(typeName, compiledTemplate, "StringHtml.html");
+  }
+
+  private void testJsonObject(final SmsRenderer.SmsTypes smsTypes) throws IOException, ParseException {
+    assertNotNull(smsTypes);
+
+    final String typeName = smsTypes.getName().toLowerCase();
+    final String fileName = typeName + "Json.json";
+    final String jsonMessageContext = this.getStringFromFile("/SmsRenderer/JSONData/" + fileName);
+
+    final JSONObject jsonObject = this.getJsonObject(jsonMessageContext);
+    assertNotNull(jsonObject);
+
+    final String compiledTemplate = this.minifyHTML(SmsRenderer.renderInBot(jsonObject, smsTypes));
```

Comment:
The minify step is duplicated three times (in `testJsonArray`, in `testJsonObject` and `testString`). All the minify steps could be put in the `verifyResultJSON`.